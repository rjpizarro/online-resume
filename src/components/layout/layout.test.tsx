import React from 'react'
import renderer from 'react-test-renderer'
import { render } from "@testing-library/react"
import * as Gatsby from 'gatsby';

import Layout from './index'

const data = {
    "site": {
      "siteMetadata": {
        "title": "Rodrigo Pizarro - Résumé"
      }
    }
}

beforeEach(() => {
  Gatsby.useStaticQuery.mockImplementation(() => (data));
});

afterEach(() => {
  Gatsby.useStaticQuery.mockImplementation(() => {});
});

describe("Layout", () => {
  it("renders normally", () => {
    const tree = renderer
      .create(<Layout>
        <span>children</span>
      </Layout>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("should call useStaticQuery", () => {
    render(
      <Layout>
        <span>children</span>
      </Layout>
    )

    expect(Gatsby.useStaticQuery).toBeCalled()
  })

  it("should render a children", () => {
    const { getByText } = render(
      <Layout>
        <span>children</span>
      </Layout>
    )

    expect(getByText("children")).toBeInTheDocument()
  })
})