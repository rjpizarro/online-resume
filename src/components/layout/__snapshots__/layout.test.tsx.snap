// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders normally 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        className="MuiTypography-root MuiTypography-h1"
        style={
          Object {
            "fontSize": 45,
            "margin": 0,
            "textAlign": "center",
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Rodrigo Pizarro - Résumé
        </a>
      </h1>
    </div>
  </header>
  <div
    style={
      Object {
        "margin": "0 auto",
        "padding": "0 1.0875rem 1.45rem",
        "paddingBottom": 200,
      }
    }
  >
    <main>
      <span>
        children
      </span>
    </main>
    <footer
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "textAlign": "center",
        }
      }
    >
      <span
        className="MuiTypography-root MuiTypography-caption"
      >
        © 
        2020
        , Built with
         
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
      </span>
      <a
        href="https://github.com/rjpizarro/resume"
        style={
          Object {
            "bottom": 0,
            "color": "#444",
            "position": "absolute",
            "right": 3,
          }
        }
      >
        <svg
          fill="currentColor"
          height={35}
          size={35}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 496 512"
          width={35}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          />
        </svg>
      </a>
    </footer>
  </div>
  <svg
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "zIndex": -10,
      }
    }
    viewBox="0 0 1440 320"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d=" M 0,32 L 48,42.7 C 96,53,192,75,288,112 C 384,149,480,203,576,197.3 C 672,192,768,128,864,117.3 C 960,107,1056,149,1152,160 C 1248,171,1344,149,1392,138.7 L 1440,128 L 1440,320 L 1392,320 C 1344,320,1248,320,1152,320 C 1056,320,960,320,864,320 C 768,320,672,320,576,320 C 480,320,384,320,288,320 C 192,320,96,320,48,320 L0,320 Z "
      fill="#00cba9"
      fillOpacity="1"
    />
  </svg>
</div>
`;
